waiter/src/waiter/process_request.clj

```diff
@@ -469,6 +469,20 @@
                     (log/warn "unable to abort as request not found inside response!"))]
       (log/info "aborted backend request:" aborted))))
 
+(defn- introspect-trailers
+  "Introspects and logs trailers received in the response"
+  [{:keys [trailers] :as response}]
+  (if trailers
+    (let [trailers-ch (async/promise-chan)
+          correlation-id (cid/get-correlation-id)]
+      (async/go
+        (when-let [trailers-data (async/<! trailers)]
+          (async/>! trailers-ch trailers-data)
+          (cid/cinfo correlation-id "response trailers:" trailers-data))
```

Comment:
Debug?

Reply:
This has me thinking that we could set up a middleware request log that logs on request completion.  These headers would be good candidates for that log.